test/io-tests/test_io.py

```diff
@@ -0,0 +1,348 @@
+    baseurl: str
+    process: object
+
+
+@pytest.fixture
+def session():
+    "Session for HTTP requests that supports connecting to unix domain sockets."
+    return requests_unixsocket.Session()
+
+
+@pytest.fixture
+def dburi():
+    "Postgres database connection URI."
+    return os.getenv("POSTGREST_TEST_CONNECTION").encode("utf-8")
+
+
+def dumpconfig(configpath, moreenv=None, stdin=None):
+    "Dump the config as parsed by PostgREST."
+    env = {**os.environ, **(moreenv or {})}
+    command = ["postgrest", "--dump-config", configpath]
+    process = subprocess.Popen(
+        command, env=env, stdin=subprocess.PIPE, stdout=subprocess.PIPE
+    )
+    process.stdin.write(stdin or b"")
+    result = process.communicate()[0]
+    process.kill()
+    process.wait()
+    if process.returncode != 0:
+        raise PostgrestError()
+    return result.decode("utf-8")
+
+
+@contextlib.contextmanager
+def run(configpath, stdin=None, moreenv=None, socket=None):
+    "Run PostgREST and yield an endpoint that is ready for connections."
+    env = {**os.environ, **(moreenv or {})}
+
+    if socket:
+        baseurl = "http+unix://" + urllib.parse.quote_plus(str(socket))
+    else:
+        baseurl = BASEURL
+
+    command = ["postgrest", configpath]
+    process = subprocess.Popen(command, stdin=subprocess.PIPE, env=env)
+
+    try:
+        process.stdin.write(stdin or b"")
+        process.stdin.close()
+
+        wait_until_ready(baseurl)
+
+        yield PostgrestProcess(baseurl=baseurl, process=process)
+    finally:
+        process.kill()
+        process.wait()
+
+
+def wait_until_ready(url):
+    "Wait for the given HTTP endpoint to return a status of 200."
+    session = requests_unixsocket.Session()
+
+    for _ in range(10):
+        try:
+            response = session.get(url, timeout=0.1)
+
+            if response.status_code == 200:
+                return
+        except requests.ConnectionError:
+            pass
+
+        time.sleep(0.1)
+
+    raise PostgrestTimedOut()
+
+
+def authheader(token):
+    "Bearer token HTTP authorization header."
+    return {"Authorization": f"Bearer {token}"}
+
+
+def jwtauthheader(claim, secret):
+    "Authorization header with signed JWT."
+    return authheader(jwt.encode(claim, secret).decode("utf-8"))
+
+
+@pytest.mark.parametrize(
+    "expectedconfig", (CONFIGSDIR / "expected").iterdir(), ids=attrgetter("name")
+)
+def test_expected_config(expectedconfig):
+    """
+    Configs as dumped by PostgREST should match an expected output.
+
+    Used to test default values, config aliases and environment variables. The
+    expected output for each file in 'configs', if available, is found in the
+    'configs/expected' directory.
+
+    """
+    expected = expectedconfig.read_text()
+    assert dumpconfig(CONFIGSDIR / expectedconfig.name) == expected
+
+
+@pytest.mark.parametrize(
+    "config",
+    [conf for conf in CONFIGSDIR.iterdir() if conf.is_file()],
```

Comment:
this iterates through all files in the `configs` directory. `io-tests.sh` only interated through files with `.config` extension. In #1691 I added a file `configs/no-defaults.env`. This is a configuration, so the `configs/` dir is right, but it sets environment variables because the files not read in as a config file.

It'd be better to use the same filtering here, I think.